src/openrct2/scripting/bindings/world/ScClimate.hpp

```diff
@@ -112,8 +116,8 @@ namespace OpenRCT2::Scripting
 
         std::shared_ptr<ScClimateState> future_get() const
         {
-            std::string weatherType = WeatherTypeToString(gClimateNext.Weather);
-            return std::make_shared<ScClimateState>(weatherType, gClimateNext.Temperature);
+            std::string weatherType = WeatherTypeToString(_gameState.ClimateNext.Weather);
+            return std::make_shared<ScClimateState>(weatherType, _gameState.ClimateNext.Temperature);
```

Comment:
```suggestion
            auto& gameState = OpenRCT2::GetGameState();
            std::string weatherType = WeatherTypeToString(gameState.ClimateNext.Weather);
            return std::make_shared<ScClimateState>(weatherType, gameState.ClimateNext.Temperature);
```
It should be like that.